Assets/Scripts/WorldEngine/World.cs

```diff
@@ -2560,43 +2569,34 @@ public void ModifyCellAltitude(TerrainCell cell, float valueOffset, float noiseF
         }
     }
 
-    private float CalculateAltitudePlusAccumulation(TerrainCell cell)
-    {
-        return cell.Altitude + cell.WaterAccumulation * OverflowFactor * RainfallToHeightConversionFactor;
-    }
-
+    /// <summary>
+    /// Adds the current buffer value to the cell's water accumulation and recalculates global maximum.
+    /// </summary>
     public void UpdateWater(TerrainCell cell)
     {
         cell.WaterAccumulation += cell.Buffer;
 
         MaxWaterAccumulation = Mathf.Max(MaxWaterAccumulation, cell.WaterAccumulation);
     }
 
-    public void PerformTerrainAlterationDrainageRegen(bool doTerrainModification = true)
+    /// <summary>
+    /// Generates drainage basins within a set of cells.
+    /// </summary>
+    public void PerformTerrainAlterationDrainageRegen()
     {
-        HashSet<TerrainCell> cellsToModify = null;
-
-        if (doTerrainModification)
+        while (_drainageEvalHeap.Count > 0)
         {
-            cellsToModify = new HashSet<TerrainCell>();
-        }
-
-        while (_drainageHeap.Count > 0)
-        {
-            TerrainCell cell = _drainageHeap.Extract(false);
-            //_cellsToDrainage.Remove(cell);
+            TerrainCell cell = _drainageEvalHeap.Extract(false);
 
             if (!cell.DrainageDone)
             {
                 bool reInsert = false;
 
-                UpdateWater(cell);
-
-                float cellAltitude = Mathf.Max(0, CalculateAltitudePlusAccumulation(cell));
+                float cellAltitude = Mathf.Max(0, cell.Altitude);
```

Comment:
Why the removal of accumulation?

Reply:
That function was not very good. It's intention was to simulate water 'overflowing' from a cell. But the effect was not really noticeable and it introduced a host of issues in the over drainage calculation logic. The drainage calculation stack is supposed to have cells partially ordered from highest to lowest, but that function messed up with that order, forcing the need to reintroduce cells into the stack, which didn't always work as intended.